reddit_decider/__init__.py

```diff
@@ -0,0 +1,345 @@
+import logging
+
+from typing import Any, Callable, Dict, Optional
+
+from baseplate import Span
+from baseplate.clients import ContextFactory
+from baseplate.frameworks.pyramid import BaseplateRequest
+from baseplate.lib import config
+from baseplate.lib.events import DebugLogger
+from baseplate.lib.events import EventLogger
+from baseplate.lib.file_watcher import FileWatcher
+from baseplate.lib.file_watcher import WatchedFileNotAvailableError
+
+import rust_decider
+
+
+logger = logging.getLogger(__name__)
+
+EMPLOYEE_ROLES = ("employee", "contractor")
+EVENT_TYPE = "expose"
+
+
+class DeciderContext:
+    """DeciderContext() is used to contain all fields necessary for
+    bucketing, targeting, and overrides.
+    DeciderContext() is populated in `make_object_for_context()`.
+    """
+
+    def __init__(
+        self,
+        user_id: str,
+        country_code: Optional[str] = None,
+        user_is_employee: Optional[bool] = None,
+        logged_in: Optional[bool] = None,
+        device_id: Optional[str] = None,
+        request_url: Optional[str] = None,
+        authentication_token: Optional[str] = None,
+        app_name: Optional[str] = None,
+        build_number: Optional[str] = None,
+        loid: Optional[str] = None,
+        cookie_created_timestamp: Optional[float] = None,
+        user_event_fields: Dict[str, Any] = None
+    ):
+        self._user_id = user_id
+        self._country_code = country_code
+        self._user_is_employee = user_is_employee
+        self._logged_in = logged_in
+        self._device_id = device_id
+        self._request_url = request_url
+        self._authentication_token = authentication_token
+        self._app_name = app_name
+        self._build_number = build_number
+        self._loid = loid
+        self._cookie_created_timestamp = cookie_created_timestamp
+        self._user_event_fields = user_event_fields
+
+    def get_user_event_fields(self):
+        return self._user_event_fields or {}
+
+    def to_dict(self):
+        return {
+            "user_id": self._user_id,
+            "country_code": self._country_code,
+            "user_is_employee": self._user_is_employee,
+            "logged_in": self._logged_in,
+            "device_id": self._device_id,
+            "request_url": self._request_url,
+            "authentication_token": self._authentication_token,
+            "app_name": self._app_name,
+            "build_number": self._build_number,
+            "loid": self._loid,
+            "cookie_created_timestamp": self._cookie_created_timestamp,
```

Comment:
no event fields?  That seems fine, if this is to be passed into decider

Reply:
actually, I should combine `self._decider_context.to_dict()` into `inputs` as well to make some of the other fields available, such as [bucketing_value](https://github.snooguts.net/reddit/reddit-service-graphql/blob/3c5b239755b8ffb5770bfaa5ef5f5fd9e5e10635/graphql-py/graphql_api/events/utils.py#L234-L236), [to the ExperimentsLogger](https://github.snooguts.net/reddit/reddit-service-graphql/blob/3c5b239755b8ffb5770bfaa5ef5f5fd9e5e10635/graphql-py/graphql_api/events/utils.py#L211-L214), since it seems everything in `input` will get dumped to the [final v2 event](https://github.snooguts.net/reddit/reddit-service-graphql/blob/3c5b239755b8ffb5770bfaa5ef5f5fd9e5e10635/graphql-py/graphql_api/events/utils.py#L295).

Reply:
removed `get_user_event_fields()` since `self._decider_context.to_dict()` contains the [3 fields that event_fields() contains](https://github.com/reddit/edgecontext/blob/42f4ee3d3e330ff70653a99a8ef5e69e88934ed6/lib/py/reddit_edgecontext/__init__.py#L291-L294)